test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -327,14 +386,19 @@ func TestVaultDynamicSecret(t *testing.T) {
 						"expected Status.LastGeneration")
 					assert.NotEmpty(t, vdsObjFinal.Status.LastRuntimePodUID)
 					assert.NotEmpty(t, vdsObjFinal.Status.LastRenewalTime)
-					assert.NotEmpty(t, vdsObjFinal.Status.SecretLease.ID)
+					if vdsObjFinal.Spec.AllowStaticCreds {
+						assert.Empty(t, vdsObjFinal.Status.SecretLease.ID)
+					} else {
+						assert.NotEmpty(t, vdsObjFinal.Status.SecretLease.ID)
+					}
+
 					assertLastRuntimePodUID(t, ctx, crdClient, operatorNS, vdsObjFinal)
-					assertDynamicSecretRotation(t, ctx, crdClient, vdsObjFinal)
+					if !vdsObjFinal.Spec.AllowStaticCreds {
+						assertDynamicSecretRotation(t, ctx, crdClient, vdsObjFinal)
```

Comment:
Silly question, maybe my brain is off today: I don't see a test for Rotation for static creds?
I see that we are checking for the existence of TTL/etc, but maybe it makes sense to test a scenario where the `lastRotated' value has changed?

Reply:
Yeah, makes sense. Adding it.

Reply:
Updated in e530736f78ad9e7a296cc53d2b0b347a2bfd1dce